ddtrace/tracer/sqlcomment.go

```diff
@@ -0,0 +1,152 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package tracer
+
+import (
+	"strconv"
+	"strings"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+)
+
+// SQLCommentInjectionMode represents the mode of SQL comment injection.
+type SQLCommentInjectionMode string
+
+const (
+	// SQLInjectionUndefined represents the comment injection mode is not set. This is the same as SQLInjectionDisabled.
```

Comment:
@gbbr : As discussed in that other thread, I updated the mode to be a string. The one tradeoff is that we lose having `disabled` be the zero value. I think it's better overall because it will make it more clear when users configure this but I added this awkward `SQLInjectionUndefined` to name the zero value. It doesn't need to be a constant (I could just replace the few usages of `SQLInjectionUndefined` with `""`) so I'm wondering if you have preferences on that one.